### PR TITLE
Labeling with image recognition model

### PR DESCRIPTION
**Pull request by briankassouf**
Uses a pre-trained [inception](https://github.com/tensorflow/models/tree/master/inception) model and tensorflow's go api to label images into one of 1,000 labels. 

Added tests and a benchmark. The GPU version is _slightly_ faster than the CPU version
```
    CPU          66703223 ns/op
    GPU          63597967 ns/op
```

### CODE REVIEW

**Comment by briankassouf** [comment]
@erulabs @pastudan

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 RUN apt-get update && apt-get install -yqq aspell aspell-en libaspell-dev tesseract-ocr tesseract-ocr-eng imagemagick optipng exiftool libjpeg-progs webp
 ADD docker/meme.traineddata /usr/share/tesseract-ocr/tessdata/meme.traineddata
 ADD docker/imagemagick_policy.xml /etc/ImageMagick-6/policy.xml
 RUN mkdir -p /etc/mandible /tmp/imagestore
 ENV MANDIBLE_CONF /etc/mandible/conf.json
-ENV GO15VENDOREXPERIMENT 1
 ADD . /go/src/github.com/Imgur/mandible
 WORKDIR /go/src/github.com/Imgur/mandible
+
+RUN wget "https://storage.googleapis.com/tensorflow/libtensorflow/libtensorflow-cpu-linux-x86_64-1.0.0.tar.gz"
```
Comment (pastudan) [inline Dockerfile]: Have you tried the GPU version? We've had pretty good luck using [open_nsfw](https://github.com/yahoo/open_nsfw)'s script with CUDA drivers, and Amazon recently released an AMI which included a bunch of ML frameworks pre-installed, making setup a bit easier... https://aws.amazon.com/marketplace/pp/B01M0AXXQB

Would be pretty sweet to get this running on all uploads.

**Comment by pastudan** [comment]
This is super cool. Code looks good. I'll try to give it a shot this coming week

File: Dockerfile (hunk 1 above)
Comment (briankassouf) [inline Dockerfile]: Yeah, good call! It will run on GPU support if run outside of docker on that AMI. I also added a `Dockerfile.gpu` that includes the CUDA libraries, it needs to be run with [nvidia-docker](https://github.com/NVIDIA/nvidia-docker) to allow it access to the host's GPUs. 

I thought it was a good idea to have a separate Dockerfile for cpu and gpu support since the gpu container does a lot of extra work. Let me know if you think they should be merged!

**Comment by briankassouf** [comment]
@pastudan I think this should be good to go. Let me know if you have any feedback/comments. Main thing to note- this probably wont return meaningful results for a lot of imgur's images due to the labels the model supports, but it's a start and another model could be swapped out in the future.

File: Dockerfile (hunk 1 above)
Comment (pastudan) [inline Dockerfile]: Hah, separate seems fine.. they do have very different dependencies

File: imageprocessor/label.go [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+	s := op.NewScope()
+	input = op.Placeholder(s, tf.String)
+	var decode tf.Output
+	switch format {
+	case "image/jpg", "image/jpeg":
+		decode = op.DecodeJpeg(s, input, op.DecodeJpegChannels(3))
+	case "image/png":
+		decode = op.DecodePng(s, input, op.DecodePngChannels(3))
+	default:
+		err = errors.New("Image type not suppored")
```
Comment (Polyrhythm) [inline imageprocessor/label.go]: :)

Comment (briankassouf) [inline imageprocessor/label.go]: Good eye :smile:

**Comment by briankassouf** [comment]
@pastudan Have you gotten a chance to take a look at this? I think it's ready to go.
